lib/modules/versioning/debian/index.ts

```diff
@@ -0,0 +1,127 @@
+import { DistroInfo, DistroInfoRecordWithVersion } from '../distro';
+import { DockerVersioningApi } from '../docker';
+import type { GenericVersion } from '../generic';
+import type { NewValueConfig, VersioningApi } from '../types';
+
+export const id = 'debian';
+export const displayName = 'Debian';
+export const urls = [
+  'https://debian.pages.debian.net/distro-info-data/debian.csv',
+];
+export const supportsRanges = false;
+
+const EOL_PROP = 'eol';
+
+class DebianVersioningApi extends DockerVersioningApi {
```

Comment:
why extending docker versioning?

Reply:
Docker versioning is the default when dealing with Debian, I just added some extra functionality to deal with codenames and rolling releases on top of it. 
so i guess instead of changing the existing docker ver api ive extended it.

so my options are either extend it as i did, or instantiating it inside the new versioning class and extending `generic versioning` instead.

another option is just duplicating the code, but i would like to avoid that approach

Reply:
can you please point out which specific docker versioning function is required for this?

Ubuntu versioning doesn't use docker versioning as base but is used as same place. so what's the difference to Debian?

Reply:
this is the default overriding, ive added the debian one.

https://github.com/renovatebot/renovate/blob/0e2440caf884a36e46903341dfd5157af55b73c8/lib/modules/manager/dockerfile/extract.ts#L140-L146


this is the default

https://github.com/renovatebot/renovate/blob/b34595c9654655fd4797158a83ecfd023277d873/lib/modules/datasource/docker/index.ts#L347-L353


so basically before this pr, debian was just another regular dep using the default docker versioning api.

Reply:
ok, but that was the same for Ubuntu and we don't extend docker versioning there, so please do the same for Debian.